### PR TITLE
First draft of the event catalog

### PR DESCRIPTION
**Pull request by medgyes**

### CODE REVIEW

File: mitzu/webapp/model.py [hunk 1]
```diff
@@ -295,3 +295,18 @@ class OnboardingFlowState:
...
+    Defines extra meta data for an event like alias or description
+
+    :param event_name: name of the event
+    :param display_name: display name of the event
+    :param description: description of the event
+    """
+
+    event_name: str
+    display_name: str
+    description: Optional[str]
```
Comment (IstvanM) [inline mitzu/webapp/model.py:312]: `= None`

File: mitzu/webapp/model.py [hunk 2]
```diff
@@ -295,3 +295,18 @@ class OnboardingFlowState:
...
+class EventMeta:
+    """
+    Defines extra meta data for an event like alias or description
+
+    :param event_name: name of the event
+    :param display_name: display name of the event
+    :param description: description of the event
+    """
+
+    event_name: str
```
Comment (IstvanM) [inline mitzu/webapp/model.py:310]: please add:
- State: Enum = oneof(Active, Inactie, possible later)
- Tags: List[String] 
- Source table: string 

- Event Name / Project is unique

File: mitzu/webapp/storage_model.py [hunk 3]
```diff
@@ -703,3 +703,36 @@ def from_model_instance(
             flow_id=state.flow_id,
             current_state=state.current_state,
         )
+
+
+class EventMetaStorageRecord(Base):
+    __tablename__ = "event_meta"
+    project_id = SA.Column(SA.String, primary_key=True)
+    event_name = SA.Column(SA.String, primary_key=True)
```
Comment (IstvanM) [inline mitzu/webapp/storage_model.py:711]: same updates as for the simple dto

File: mitzu/webapp/model.py (hunk 1 above)
Comment (medgyes) [reply mitzu/webapp/model.py]: https://github.com/mitzu-io/mitzu/pull/241/commits/aad4379e8ab5de6ebc6176cdecf30bb5ffe76395

File: mitzu/webapp/pages/explore/event_segment_handler.py [hunk 4]
```diff
@@ -21,15 +23,15 @@ def create_event_name_dropdown(
...
-        [
-            {"label": value_to_label(v), "value": v}
-            for v in discovered_project.get_all_event_names()
-        ]
-        if discovered_project is not None
-        else []
-    )
+    options: List[Dict] = []
+    for v in discovered_project.get_all_event_names():
+        if v in event_catalog.keys():
```
[the remark this answers is not in the document]
Comment (medgyes) [reply mitzu/webapp/pages/explore/event_segment_handler.py]: https://github.com/mitzu-io/mitzu/pull/241/commits/aad4379e8ab5de6ebc6176cdecf30bb5ffe76395

File: mitzu/webapp/pages/manage_event_catalog.py [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+def no_project_layout():
+    return layout(None)
+
+
+@restricted_layout
+def layout(project_id: Optional[str], **query_params) -> bc.Component:
+    storage = DEPS.Dependencies.get().storage
+
+    projects = storage.list_projects()
+    # selected_project: Optional[M.Project] = None
```
[the remark this answers is not in the document]
Comment (medgyes) [reply mitzu/webapp/pages/manage_event_catalog.py]: https://github.com/mitzu-io/mitzu/pull/241/commits/dc762c50460d1c86c2bf6d5db722f62c6ecaf5fa

**Review by IstvanM — APPROVED** [review]

**Pull request merged**